src/main/java/core/basesyntax/serviceimpl/WriterServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.service.WriterService;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class WriterServiceImpl implements WriterService {
+    private static final String REPORT_HEADER = "fruit,quantity";
+
+    @Override
+    public void writeToFile(String filePath, List<String> lines) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            writer.write(REPORT_HEADER);
+            writer.newLine();
+            for (String line : lines) {
+                writer.write(line);
+                writer.newLine();
+            }
+        } catch (IOException e) {
```

Comment:
Exception handling is done correctly by rethrowing a RuntimeException with an informative message and the original exception.